dashboard/user.py

```diff
@@ -1,25 +1,36 @@
-import hashlib
 import requests
-from utils import get_secret
+from dashboard.utils import get_secret
 
 """User class that governs maniuplation of session['userdata']"""
 
 
 class User(object):
     def __init__(self, session):
         """Constructor takes user session."""
-        self.userinfo = session['userinfo']
+        self.userinfo = session.get('userinfo', None)
+        self.api_token = None
 
-    def userhash(self):
-        """Represent user e-mail as hex hash."""
-        m = hashlib.md5()
-        m.update(self.userinfo['email'])
-        return m.hexdigest()
+    def apps(self, app_list):
+        """Return a list of the apps a user is allowed to see in dashboard."""
+        authorized_apps = {
+            'apps': []
+        }
+
+        for app in app_list['apps']:
+            if self._is_valid_yaml(app):
+                if self._is_authorized(app):
+                    authorized_apps['apps'].append(app)
```

Comment:
super nit: a bit more pythonic version of this could be

```python
return [app for app in app_list['apps'] if self._is_authorized(app) and self._is_valid_yaml(app)]
```